### ISSUE TITLE
Bypass JSR107 layer when creating cache cause it to be not found

### ISSUE THREAD

**Issue by henri-tremblay** [ehcache/ehcache3#2494]
If a cache is created directory on the wrapped Ehcache CacheManager, the JSR107 won' see it. See this code:

```java
    @Test
    public void test() {
        CacheManager cacheManager = Caching.getCachingProvider().getCacheManager();
        org.ehcache.CacheManager ecm = cacheManager.unwrap(org.ehcache.CacheManager.class);
        ecm.createCache("foo", CacheConfigurationBuilder.newCacheConfigurationBuilder(Integer.class, String.class, ResourcePoolsBuilder.heap(1)));

        Cache<Integer, String> cache = cacheManager.getCache("foo", Integer.class, String.class);
        assertThat(cache).isNotNull(); // fails
    }
```

However, add a `cacheManager.getCacheNames()` before calling `getCache` repairs it since it causes a cache refresh.

I think it should work or at least be coherent.

**Comment by chrisdennis** [comment]
There has been work in this area recently... I think this might be fixed. This should get reassessed in-house and closed if it really has been fixed.

**Comment by nnares** [comment]
@chrisdennis, I have validated the mentioned use case and found that issue is still there (sample JUnit is still failing)

The above code is logically correct, as the cache is been created post ` unwrap()` method call, So it should not be visible to javaxCacheManager.
Current implement allows visibility of such new cache which has been created post ` unwrap()` method call after a cache refresh, which can be achieved by calling - `cacheManager.getCacheNames() `
My stand here is logged issue is there, but the mentioned behavior is logically correct.
If we still want a fix for this, we need to add cache refresh logic there inside of `.getCache()` method

**Comment by nnares** [comment]
Fixed(daggy) in #3117, #3119
